lib/hiera/backend/vault_backend.rb

```diff
@@ -92,22 +106,29 @@ def lookup_generic(key, scope)
           end
 
           return nil if secret.nil?
-
```

Comment:
Instead of sprinkling `if @config[:kv_version] == 2` throughout the following, couldn't you just create a variable here to represents secret.data in the kv1 case, and secret.data[:data] in the kv2 case, and work off of that from here on out? (This is already a pretty ugly collection of conditionals, seems a shame to make it worse.) ie:
```
sd = @config[:kv_version] == 2 ? secret.data[:data] : secret.data
```
and then reference `sd` instead of `secret.data` everywhere, without changing any of the existing logic?